core/src/main/scala/kafka/log/Log.scala

```diff
@@ -582,6 +582,64 @@ class Log(val dir: File,
     }
   }
 
+  private class SizeBasedRetentionCheck(var diff: Long) {
```

Comment:
Because the shouldDelete function is used in a couple of places and I didn't want to duplicate the logic. Also couldn't think of another way of not creating the same function twice as it is modifying a var - i'm open to suggestions